### ISSUE TITLE
Ability to customize network rank LoRA dreambooth training

### ISSUE THREAD

**Issue by isidentical** [huggingface/diffusers#3944]
**Is your feature request related to a problem? Please describe.**
For LoRA training, custom UIs like A1111 or kohya SD allow customization of the LoRA network rank which can make the results look much nicer but with the side effect of making LoRAs larger.

**Describe the solution you'd like**
Allowing configuration of LoRA network rank on the training dreambooth script (https://github.com/isidentical/diffusers/blob/4aca45500cf601b2bb4769651d3081cac845312d/examples/dreambooth/train_dreambooth_lora.py)

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by hsebik** [comment]
Could you help me to change rank parameter? In my training, I added new parameter --rank=128 but training result again outputs 3.1 mb model file. Also in the script of diffuser/src/diffusers/models/attention_processor.py, I have changed rank=4 to 128 again nothing is changed.

**Comment by sayakpaul** [comment]
Could you share your command script?

**Comment by hsebik** [comment]
python /diffusers/examples/dreambooth/train_dreambooth_lora.py --pretrained_model_name_or_path="SG161222/Realistic_Vision_V4.0" --instance_data_dir=$Datadir --dataloader_num_workers=8 --resolution=512 --train_batch_size=2 --num_train_epochs=4 --instance_prompt="a photo of qwyz" --max_train_steps=1200 --learning_rate=1e-4 --lr_scheduler="cosine" --lr_warmup_steps=0 --output_dir=$Outdir  --checkpointing_steps=600 --validation_prompt="a photo of qwyz smiling" --validation_epochs=100 --rank=128 --seed=0 --mixed_precision fp16

**Comment by isidentical** [comment]
In regards to the usage, this is what we use (and we can successfully get much larger LoRA files, over 100mbs+ vs only 3mbs with the default rank)

```
    subprocess.check_call(
        [
            "accelerate",
            "launch",
            prepare_training_script(),
            "--pretrained_model_name_or_path",
            model_name,
            "--pretrained_vae_model_name_or_path",
            custom_vae_name,
            "--instance_data_dir",
            inputs_dir,
            "--output_dir",
            outputs_dir,
            "--instance_prompt",
            instance_prompt,
            "--resolution",
            "512",
            "--rank",
            rank,
            "--train_batch_size",
            "1",
            "--gradient_accumulation_steps",
            "1",
            "--checkpointing_steps",
            "100",
            "--learning_rate",
            "1e-4",
            "--lr_scheduler",
            "constant",
            "--lr_warmup_steps",
            "0",
            "--max_train_steps",
            max_training_steps,
            "--seed",
            seed,
            "--train_text_encoder",
            "--resume_from_checkpoint",
            "latest",
        ]
    )
```
